scipy/spatial/distance.py

```diff
@@ -861,6 +861,9 @@ def jaccard(u, v, w=None):
     0.66666666666666663
 
     """
+    if np.isin(u, [0, 1]).all() == False or np.isin(v, [0, 1]).all() == False:
+        raise ValueError('u and v should be boolean 1-D arrays')
```

Comment:
Any input validation should probably come after the `_validate_vector()` calls to minimize data copying.

Note that the code that follows this uses `u != 0` and `v != 0` to coerce the values to `bool` `True`s and `False`s, so it was written to expect numerical values other than strict 0s and 1s. I would probably fix this bug by explicitly casting these arrays to `bool` arrays (and removing the superfluous `!= 0`s) rather than doing relatively expensive checking for exact 0s and 1s. I think that comports with the intended functionality and documentation best.